src/app/configuration/csv-policy-import-dialog/csv-policy-import-dialog.component.ts

```diff
@@ -0,0 +1,168 @@
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { ConsentService } from 'src/app/consent/consent.service';
+import { take } from 'rxjs/operators';
+import { TranslateService } from '@ngx-translate/core';
+
+@Component({
+  selector: 'app-csv-policy-import-dialog',
+  templateUrl: './csv-policy-import-dialog.component.html',
+  styleUrls: ['./csv-policy-import-dialog.component.css']
+})
+export class CsvPolicyImportDialogComponent {
+  errorMessages: string[] = [];
+  isLoading: boolean = false;
+  selectedFiles: File[] = [];
+  selectedFileNames: string[] = [];
+  fileContents: string[] = [];
+  importResult: { added: number; failed: number } | null = null;
+  totalRows: number = 0;
+  fileValid: boolean = true;
+
+  constructor(
+    public dialogRef: MatDialogRef<CsvPolicyImportDialogComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: { policySetId: string },
+    private consentService: ConsentService,
+    private translate: TranslateService,
+  ) {}
+
+  onFileSelected(event: any) {
+    const files: FileList = event.target.files;
+    for (let i = 0; i < files.length; i++) {
+      const file = files[i];
+      if (!this.selectedFiles.some(f => f.name === file.name)) {
+        this.handleFile(file);
+      }
+    }
+  }
+
+  onDrop(event: DragEvent) {
+    event.preventDefault();
+    if (event.dataTransfer && event.dataTransfer.files.length > 0) {
+      const files = event.dataTransfer.files;
+      for (let i = 0; i < files.length; i++) {
+        const file = files[i];
+        if (!this.selectedFiles.some(f => f.name === file.name)) {
+          this.handleFile(file);
+        }
+      }
+    }
+  }
+  
+  onDragOver(event: DragEvent) {
+    event.preventDefault();
+  }
+
+ handleFile(file: File) {
+  const reader = new FileReader();
+  reader.onload = (e: any) => {
+    const fileContent = e.target.result;
+    const policies = this.parseCSV(fileContent);
+    const validationErrors = this.validatePolicies(policies);
+
+    if (validationErrors.length > 0) {
+      const groupedErrors: { [key: string]: string } = {};
+      validationErrors.forEach(err => {
+        if (!groupedErrors[err.category]) {
+          groupedErrors[err.category] = `${file.name}: ${err.message}`;
+        }
+      });
+      this.errorMessages = Object.values(groupedErrors);
+      this.fileValid = false;
+      return;
+    }
+
+    this.selectedFiles.push(file);
+    this.selectedFileNames.push(file.name);
+    this.fileContents.push(fileContent);
+    this.errorMessages = [];
+    this.fileValid = true;
+  };
+    reader.readAsText(file);
+  }
+
+  removeFile(fileName: string) {
+    const index = this.selectedFiles.findIndex(f => f.name === fileName);
+    if (index !== -1) {
+      this.selectedFiles.splice(index, 1);
+      this.selectedFileNames.splice(index, 1);
+      this.fileContents.splice(index, 1);
+    }
+  }
+
+  parseCSV(csvText: string): Array<{ code: string, name: string }> {
```

Comment:
you can replace duplicated functions in this file and in policy-set-form.component.ts with a shared service implementation